File: benchmarks/poly/sortnums/sortnums.lua
local function main()
    local _t0 = os.clock()
    local n = 1000000
    local seed = 12345
    local arr = {}
    for i = 1, n do
        seed = (48271 * seed) % 2147483647
        arr[i] = seed % 1000000
    end

    table.sort(arr)

    local checksum = 0
    for i = 1, n do
        checksum = (checksum * 31 + arr[i]) % 1000000007
    end

    print(arr[1])
    print(arr[n])
    print(checksum)
    io.stderr:write(string.format("bench_ns=%.0f\n", (os.clock() - _t0) * 1e9))
end

main()

File: benchmarks/poly/sortnums/sortnums.rs
fn main() {
    let t0 = std::time::Instant::now();
    let n = 1_000_000i64;
    let mut seed: i64 = 12345;
    let mut arr: Vec<i64> = Vec::with_capacity(n as usize);
    for _ in 0..n {
        seed = (48271 * seed) % 2147483647;
        arr.push(seed % 1000000);
    }

    arr.sort();

    let mut checksum: i64 = 0;
    for v in &arr {
        checksum = (checksum * 31 + v) % 1000000007;
    }

    println!("{}", arr[0]);
    println!("{}", arr[(n - 1) as usize]);
    println!("{}", checksum);
    eprintln!("bench_ns={}", t0.elapsed().as_nanos());
}
